blockchain/upgrade_test.go

```diff
@@ -343,3 +348,132 @@ func TestBlockIndexSerializationV2(t *testing.T) {
 		}
 	}
 }
+
+// TestBlockIndexSerializationV3 ensures serializing block index entries in the
+// version 3 format works as expected.
+func TestBlockIndexSerializationV3(t *testing.T) {
+	t.Parallel()
+
+	// base data is based on block 150287 on mainnet and serves as a template
+	// for the various tests below.
+	baseHeader := wire.BlockHeader{
+		Version:      4,
+		PrevBlock:    *mustParseHash("000000000000016916671ae225343a5ee131c999d5cadb6348805db25737731f"),
+		MerkleRoot:   *mustParseHash("5ef2bb79795d7503c0ccc5cb6e0d4731992fc8c8c5b332c1c0e2c687d864c666"),
+		StakeRoot:    *mustParseHash("022965059b7527dc2bc18daaa533f806eda1f96fd0b04bbda2381f5552d7c2de"),
+		VoteBits:     0x0001,
+		FinalState:   hexToFinalState("313e16e64c0b"),
+		Voters:       4,
+		FreshStake:   3,
+		Revocations:  2,
+		PoolSize:     41332,
+		Bits:         0x1a016f98,
+		SBits:        7473162478,
+		Height:       150287,
+		Size:         11295,
+		Timestamp:    time.Unix(1499907127, 0),
+		Nonce:        4116576260,
+		ExtraData:    hexToExtraData("8f01ed92645e0a6b11ee3b3c0000000000000000000000000000000000000000"),
+		StakeVersion: 4,
+	}
+	baseVoteInfo := []blockIndexVoteVersionTuple{
+		{version: 4, bits: 0x0001},
+		{version: 4, bits: 0x0015},
+		{version: 4, bits: 0x0015},
+		{version: 4, bits: 0x0001},
+	}
+
+	// Version 3 block status flags used in the tests.
+	const (
+		v3StatusDataStored = 1 << 0
+		v3StatusValidated  = 1 << 1
+	)
```

Comment:
Unless there is a reason for avoiding a global const block consider it for consistency.

Reply:
Disregard.